torch/utils/tensorboard/pytorch_graph.py

```diff
@@ -0,0 +1,263 @@
+            if hasattr(node, 'input_or_output'):
+                self.uniqueNameToScopedName[key] = node.input_or_output + '/' + node.uniqueName
+            if hasattr(node, 'scope'):
+                if node.scope == '' and self.shallowestScopeName:
+                    self.uniqueNameToScopedName[node.uniqueName] = self.shallowestScopeName + '/' + node.uniqueName
+        # replace name
+        # print(self.uniqueNameToScopedName)
+        for key, node in self.nodes_IO.items():
+            self.nodes_IO[key].inputs = [self.uniqueNameToScopedName[node_input_id] for node_input_id in node.inputs]
+            if node.uniqueName in self.uniqueNameToScopedName:
+                self.nodes_IO[key].uniqueName = self.uniqueNameToScopedName[node.uniqueName]
+
+    def to_proto(self):
+        import numpy as np
+        nodes = []
+        node_stats = []
+        for v in self.nodes_IO.values():
+            nodes.append(Node_proto(v.uniqueName,
+                                    input=v.inputs,
+                                    outputsize=v.tensorSize,
+                                    op=v.kind,
+                                    attributes=v.attributes))
+
+            if v.tensorSize and len(v.tensorSize) > 0:  # assume data is float32, only parameter is counted
+                node_stats.append(
+                    NodeExecStats(node_name=v.uniqueName,
+                                  all_start_micros=int(time.time() * 1e7),
+                                  all_end_rel_micros=42,
+                                  memory=[AllocatorMemoryUsed(allocator_name="cpu",
+                                                              total_bytes=int(np.prod(v.tensorSize)) * 4)]))
+
+        return nodes, node_stats
+
+
+# one argument: 'hasAttribute', 'hasAttributes',
+def parse(graph, args=None, omit_useless_nodes=True):
+    n_inputs = len(args)  # not sure...
+
+    scope = {}
+    nodes_py = Graph_py()
+    for i, node in enumerate(graph.inputs()):
+        if omit_useless_nodes:
+            if len(node.uses()) == 0:  # number of user of the node (= number of outputs/ fanout)
+                continue
+
+        if i < n_inputs:
+            nodes_py.append(Node_py_IO(node, 'input'))
+        else:
+            nodes_py.append(Node_py_IO(node))  # parameter
+
+    for node in graph.nodes():
+        nodes_py.append(Node_py_OP(node))
+
+    for node in graph.outputs():  # must place last.
+        Node_py_IO(node, 'output')
+    nodes_py.findCommonRoot()
+    nodes_py.populate_namespace_from_OP_to_IO()
+    return nodes_py.to_proto()
+
+
+def graph(model, args, verbose=False, **kwargs):
+    import torch
+
+    def _optimize_trace(trace, operator_export_type):
+        trace.set_graph(_optimize_graph(trace.graph(), operator_export_type))
+
+    def _optimize_graph(graph, operator_export_type):
+        # torch._C._jit_pass_remove_inplace_ops(graph)
```

Comment:
Let's make sure the `try: ... except:` just brought into tensorboardX also gets added here.